tests/fixture/tmpnet/node.go

```diff
@@ -323,6 +319,59 @@ func (n *Node) GetUniqueID() string {
 	return n.network.UUID + "-" + strings.ToLower(nodeIDString[startIndex:endIndex])
 }
 
+// composeFlags determines the set of flags that should be used to
+// start the node.
+func (n *Node) composeFlags() (FlagsMap, error) {
+	flags := maps.Clone(n.Flags)
+
+	// Apply the network defaults first so that they are not overridden
+	flags.SetDefaults(n.network.DefaultFlags)
+
+	flags.SetDefaults(DefaultTmpnetFlags())
+
+	// Convert the network id to a string to ensure consistency in JSON round-tripping.
+	flags.SetDefault(config.NetworkNameKey, strconv.FormatUint(uint64(n.network.GetNetworkID()), 10))
+
+	// Set the bootstrap configuration
+	bootstrapIPs, bootstrapIDs := n.network.GetBootstrapIPsAndIDs(n)
+	flags.SetDefault(config.BootstrapIDsKey, strings.Join(bootstrapIDs, ","))
+	flags.SetDefault(config.BootstrapIPsKey, strings.Join(bootstrapIPs, ","))
+
+	// TODO(marun) Maybe avoid computing content flags for each node start?
+
+	if n.network.Genesis != nil {
+		genesisFileContent, err := n.network.GetGenesisFileContent()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get genesis file content: %w", err)
+		}
+		flags.SetDefault(config.GenesisFileContentKey, genesisFileContent)
+
+		isSingleNodeNetwork := (len(n.network.Nodes) == 1 && len(n.network.Genesis.InitialStakers) == 1)
```

Comment:
nit unneeded parentheses

```suggestion
		isSingleNodeNetwork := len(n.network.Nodes) == 1 && len(n.network.Genesis.InitialStakers) == 1
```